src/services/historyService.js

```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+
+const addToHistory = (dispatch, video, token) => {
+  try {
+    (async () => {
+      const {
+        data: { history },
+      } = await axios.post(
+        "/api/user/history",
+        {
+          video,
+        },
+        {
+          headers: {
+            authorization: token,
+          },
+        }
+      );
+      history &&
+        dispatch({
+          type: "SET_HISTORY",
+          payload: history,
+        });
+    })();
+  } catch (error) {
+    console.log("Error in add to history handler", error);
```

Comment:
better to handle this with a throw statement, or can be improved with a toast implementation in the future